inc/Integrations/Google/Sheets/GoogleSheetsDataSource.php

```diff
@@ -100,6 +100,14 @@ public static function preprocess_get_response( array $response_data, array $inp
 		if ( isset( $response_data['values'] ) && is_array( $response_data['values'] ) ) {
 			$values = $response_data['values'];
 			$columns = array_shift( $values ); // Get column names from first row
+
+			// if the values are now empty, give back $selected_row as null
+			// This can happen if rows are deleted in the sheet after they
+			// have been used in a remote data block.
+			if ( empty( $values ) ) {
```

Comment:
I noticed that when you delete a row in a google sheet that's already been mapped, it causes the whole plugin to error out due to an out of bounds check here. I ran into it while trying to test out what would happen if I did this case. It will render the stale data now.